packages/eddsa-pcd/src/EdDSAPCD.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  DisplayOptions,
+  PCD,
+  PCDPackage,
+  SerializedPCD,
+  StringArgument,
+  StringArrayArgument
+} from "@pcd/pcd-types";
+// eslint-disable-next-line @typescript-eslint/triple-slash-reference
+/// <reference path="util/declarations/circomlibjs.d.ts" />
+import { buildEddsa, buildPoseidon } from "circomlibjs";
+import { v4 as uuid } from "uuid";
+import { EdDSACardBody } from "./CardBody";
+
+export const EdDSAPCDTypeName = "eddsa-pcd";
+
+export interface EdDSAPCDArgs {
+  privateKey: StringArgument;
+  message: StringArrayArgument;
+  id: StringArgument;
+}
+
+export interface EdDSAPCDClaim {
+  publicKey: [string, string];
+  message: bigint[];
+}
+
+export interface EdDSAPCDProof {
+  signature: string;
+}
+
+export class EdDSAPCD implements PCD<EdDSAPCDClaim, EdDSAPCDProof> {
+  type = EdDSAPCDTypeName;
+  claim: EdDSAPCDClaim;
+  proof: EdDSAPCDProof;
+  id: string;
+
+  public constructor(id: string, claim: EdDSAPCDClaim, proof: EdDSAPCDProof) {
+    this.id = id;
+    this.claim = claim;
+    this.proof = proof;
+  }
+}
+
+const fromHexString = (hexString: string) => Buffer.from(hexString, "hex");
+
+const toHexString = (bytes: Uint8Array) => Buffer.from(bytes).toString("hex");
+
+export async function prove(args: EdDSAPCDArgs): Promise<EdDSAPCD> {
+  if (!args.privateKey.value) {
+    throw new Error("No private key value provided");
+  }
+
+  if (!args.message.value) {
+    throw new Error("No message value provided");
+  }
+
+  let message;
+  try {
+    message = args.message.value.map((fieldStr: string) => BigInt(fieldStr));
+  } catch (e) {
+    throw new Error("Could not convert message contents to bigint type");
+  }
+
+  const id = typeof args.id.value === "string" ? args.id.value : uuid();
+
+  const prvKey = fromHexString(args.privateKey.value);
+
+  const eddsa = await buildEddsa();
+  const poseidon = await buildPoseidon();
```

Comment:
there's a good chance this leaks resources if it's invoked every time you call `prove`. see the ethereum group signature pcd for an example of how we initialize required libraries once for the package so that doesn't have to be done on each `prove` and `verify`: https://github.com/proofcarryingdata/zupass/blob/main/packages/ethereum-group-pcd/src/EthereumGroupPCD.ts#L108-L123